src/nvim/decoration.h

```diff
@@ -58,11 +58,24 @@ typedef struct {
   int attr_id;  // cached lookup of decor.hl_id
   bool virt_text_owned;
   int win_col;
+  uint64_t ns_id;
+  uint64_t mark_id;
 } DecorRange;
 
+typedef struct {
```

Comment:
we don't need an extra struct and extra `active_lookless` array. Add an extra field for Decoration like `bool external` instead.

Reply:
done.